### PR TITLE
Get Spraynozzle to do its own load balancing

### PR DESCRIPTION
**Pull request by dfellis**
cc @depombo @isaacbrodsky

This is a simplistic load-balancing system that has an implicit lock via incrementing a static integer. There *will* be multiple messages sent to the same box at the same time this way, but it should be small. We can improve this load-balancing over time.

### CODE REVIEW

**Comment by dfellis** [comment]
Oh, I forgot to mention, this *should* be backwards compatible with current usage.

**Comment by isaacbrodsky** [comment]
Two comments on KafkaPoster:

- It looks like if a classifier is down then some percentage of data will be dropped. Do we want to redirect traffic if a classifier host is down?
- Make `currentUrl` non-static?

**Comment by dfellis** [comment]
On point 1: Yes, we should. I wanted to do that in a separate diff after we prove that we can handle more traffic by bypassing HAProxy.

On point 2: I made it static so all of the instances share the same integer value for the current URL index. This should prevent it from doing hundreds of requests all on the first box, then the second box, etc.

**Comment by isaacbrodsky** [comment]
:+1:

**Comment by depombo** [comment]
This looks like a good to me, other than the crazy long zk path

File: src/main/java/com/uber/kafkaSpraynozzle/KafkaSpraynozzle.java [hunk 1]
```diff
@@ -82,14 +85,14 @@ public static void main(String[] args) throws Exception {
 
         if (!buffering) {
             // Clear out zookeeper records so the spraynozzle drops messages between runs
-            clearZkPath(zkClient, "/consumers/kafka_spraynozzle_" + topics[0] + cleanedUrl);
+            clearZkPath(zkClient, "/consumers/kafka_spraynozzle_" + topics[0] + cleanedUrls);
```
Comment (depombo) [inline src/main/java/com/uber/kafkaSpraynozzle/KafkaSpraynozzle.java]: lets ddos zk

**Pull request merged**